src/daemon.rs

```diff
@@ -148,11 +148,14 @@ impl Daemon {
     }
 
     pub(crate) fn estimate_fee(&self, nblocks: u16) -> Result<Option<Amount>> {
-        Ok(self
-            .rpc
-            .estimate_smart_fee(nblocks, None)
-            .context("failed to estimate fee")?
-            .fee_rate)
+        let res = self.rpc.estimate_smart_fee(nblocks, None);
+        if let Err(bitcoincore_rpc::Error::JsonRpc(jsonrpc::Error::Rpc(ref e))) = res {
```

Comment:
I think you can use pattern matching here, like
```rust
if let Err(bitcoincore_rpc::Error::JsonRpc(jsonrpc::Error::Rpc(RpcError { code: -32603, .. }))) = res {
    ...
} else {
    ...
}
```

Reply:
Good catch, thanks!